addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/ws/datatypes/WSBaseDataType.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.ihc.internal.ws.datatypes;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.util.Iterator;
+
+import javax.xml.namespace.NamespaceContext;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+
+import org.openhab.binding.ihc.internal.ws.exeptions.IhcExecption;
+import org.xml.sax.InputSource;
+
+/**
+ * Base data class for all controllers data values.
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+public abstract class WSBaseDataType {
+
+    static public String parseXMLValue(String xml, String xpathExpression) throws IhcExecption {
+        InputStream is;
+        try {
+            is = new ByteArrayInputStream(xml.getBytes("UTF8"));
```

Comment:
Close this stream, use try with auto close as mentioned in other review comment.

Reply:
Done